app/views/articles/index.html.haml

```diff
@@ -61,10 +59,24 @@
         %td= article.created_at
         %td= image_tag article.image.url(:thumb)
         %td= link_to 'Show', article_path(article)
-        %td= link_to 'Edit', edit_article_path(article)
-        %td= link_to 'Destroy', article_path(article), |
-            method: :delete,                          |
-            data: { confirm: 'Are you sure?' }        |
+        - if can? :update, Article
+          - if (current_user.id == article.user_id and current_user.role.name == 'author') or current_user.role.name == 'admin'
+            %td= link_to 'Edit', edit_article_path(article)
+          -else
+            %td= "Restricted Author Access"
```

Comment:
Generic comment over UX, when someone doesn't know that he isn't authorized because of certain unknown rules, he shouldn't be told inside the page, that this specific place, he isn't authorized. This is really 90's old style.